osu.Game/OsuGame.cs

```diff
@@ -794,21 +794,21 @@ protected override void UpdateAfterChildren()
 
         protected virtual void ScreenChanged(IScreen current, IScreen newScreen)
         {
-            switch (newScreen)
-            {
-                case Intro intro:
-                    introScreen = intro;
-                    break;
-
-                case MainMenu menu:
-                    menuScreen = menu;
-                    break;
-            }
-
             if (newScreen is IOsuScreen newOsuScreen)
             {
                 OverlayActivationMode.Value = newOsuScreen.InitialOverlayActivationMode;
 
+                switch (newOsuScreen)
```

Comment:
Not sure if this is relevant but isn't this changing the logic?
I think your goal was to delay the assignment happening in this switch but shouldn't it be placed outside of the condition like it was before (read: after the condition)?

Reply:
My goal was to make sure the `OverlayActivationMode` value above gets set before the `break` is hit in the Intro in the line above.